worker/migrationminion/worker_test.go

```diff
@@ -204,13 +208,30 @@ func (s *Suite) TestVALIDATIONCantConnect(c *gc.C) {
 	c.Assert(err, jc.ErrorIsNil)
 	defer workertest.CleanKill(c, w)
 
+	// Advance time enough for all of the retries to be exhausted.
+	for i := 0; i < 10; i++ {
+		// Don't worry about whether there are any waiters - we'll
+		// get them next time.
+		s.clock.Advance(time.Second)
+		time.Sleep(coretesting.ShortWait)
```

Comment:
This does leave the potential for a race on slow substrates as we are assuming the worker is working at a fast enough speed.

Would be better to use wait advance the correct number of times IMO, and even have a channel that is closed by the APIOpen func above after 10 attempts. That gives a definitive thing to wait on that will ensure that the test passes.

Reply:
Yeah, will change this - it was basically just laziness to avoid working out the actual sleep times, but I guess I can have the computer calculate those for me!